view/src/components/produce.js

```diff
@@ -0,0 +1,724 @@
+import React, { Component } from "react";
+
+import withStyles from "@material-ui/core/styles/withStyles";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Slide from "@material-ui/core/Slide";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import Container from "@material-ui/core/Container";
+import CardActions from "@material-ui/core/CardActions";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import CardContent from "@material-ui/core/CardContent";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+import InputBase from "@material-ui/core/InputBase";
+import SearchIcon from "@material-ui/icons/Search";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import Box from "@material-ui/core/Box";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
+
+const styles = (theme) => ({
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+  },
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  submitButton: {
+    display: "block",
+    color: "white",
+    textAlign: "center",
+    position: "absolute",
+    top: 14,
+    right: 10,
+  },
+  floatingButton: {
+    position: "fixed",
+    bottom: 0,
+    right: 0,
+  },
+  form: {
+    width: "98%",
+    marginLeft: 13,
+    marginTop: theme.spacing(3),
+  },
+  toolbar: theme.mixins.toolbar,
+  root: {
+    minWidth: 470,
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  uiProgess: {
```

Comment:
maybe this can be uiProgress instead :P

Reply:
oh wait i just realized this is merged - ill change this in my branch instead. i'm still figuring out the spacing changes btw